lib/pbench/server/api/resources/query_apis/metadata_index/iteration_samples.py

```diff
@@ -0,0 +1,419 @@
+from http import HTTPStatus
+from logging import Logger
+from typing import AnyStr, Dict, List
+
+from flask import Response, jsonify
+from flask_restful import abort
+
+from pbench.server import PbenchServerConfig
+from pbench.server.api.resources import (
+    JSON,
+    Parameter,
+    ParamType,
+    PostprocessError,
+    Schema,
+)
+from pbench.server.api.resources.query_apis import CONTEXT
+from pbench.server.api.resources.query_apis.metadata_index import MetadataBase
+from pbench.server.database.models.template import Template, TemplateNotFound
+
+
+class IterationSampleNamespace(MetadataBase):
+    """
+    Iteration samples API that returns namespace for each whitelisted
+    result-data-sample subdocuments as well as lists of available values for
+    each name.
+
+    This class inherits the common IterationSamples class and builds an
+    aggregated term query filtered by user supplied run id.
+    """
+
+    WHITELIST_AGGS_FIELDS = ["run", "sample", "iteration", "benchmark"]
+
+    def __init__(self, config: PbenchServerConfig, logger: Logger):
+        super().__init__(
+            config,
+            logger,
+            Schema(Parameter("run_id", ParamType.STRING, required=True)),
```

Comment:
I think it will accept an empty string as a `run_id` but it will abort with FORBIDDEN status (since Dataset with empty run id will not be not be found)

Reply:
Did you mean `NOT_FOUND`, rather than `FORBIDDEN`?  (I'm looking at `RunIdBase.preprocess()` line 64 in `lib/pbench/server/api/resources/query_apis/metadata_index/__init__.py`.)

Reply:
Ah that is correct, yes it would return `NOT_FOUND`